README.md

```diff
@@ -7,3 +7,35 @@ slog is a library for capturing structured log information. In contrast to "trad
 * captures arbitrary key-value metadata on each log event
 
 slog forwards messages to [`log`](https://golang.org/pkg/log/) by default. But you probably want to write a a custom output to make use of the context and metadata. At [Monzo](https://monzo.com/), slog captures events both on a per-service and a per-request basis (using the context information) and sends them to a centralised logging system. This lets us view all the logs for a given request across all the micro-services it touches.
+
+## Usage
+
+Internally at Monzo, we recommend that users always prefer structured logging where possible. An example of using slog for this would be:
+
+```go
+slog.Info(ctx, "Loading widget", map[string]interface{}{
+    "stage": "reticulating splines",
+})
+```
+
+When logging an error, we recommend attaching the actual error in the "error" key:
+
+```go
+err := errors.New("No splines to reticulate")
+slog.Error(ctx, "Failed to load widget", map[string]interface{}{
+    "error": err,
+})
```

Comment:
Could slog not do this for you automatically? I.e. if any param satisfies the error interface then it adds it to the params unless the params already contain an `error` field. This is somewhat similar to the params merging it already does: https://github.com/monzo/slog/blob/002e22b2c169007410273e08842fe5bef480fef9/event.go#L87-L96

Reply:
Yep we can absolutely do this (I actually have it in a separate PR), but it means picking a convention for errors explicitly, and this PR is just around the metadata capability.